### PR TITLE
Fix and enhance RTCP stats and link quality metrics.

### PR DESCRIPTION
**Pull request by atoppi**
This PR heavily refactors the way Janus calculates some RTCP stats, estimate link metrics and generate receiver reports.

- Before this PR the lost count was basically incremented before sending the first NACK for a packet.
Now the calculation happens whenever a RTCP Receiver Report is being sent, that is once in a second.
The cumulative loss is now obtained as the accumulation of the _packets lost_ in every 1 second time window.
That is, for every RR being sent:
```
lost_interval = expected_interval - received_interval
cumulative_loss += lost_interval
```

- The `expected` packets semantics has not been changed and is calculated from the highest sequence number in the stream.

- The `received` packet counter is now incremented upon the reception of one of the followings:

1. in-order RTP media packet
2. out-of-order RTP media packet for RFC4588 enabled RTP streams
3. out-of-order, not too late (*), RTP media packet for non-RFC4588-enabled and NACKs-enabled RTP streams

- The `retransmitted` packet counter  is now incremented upon the reception of one of the followings:

1. RTP RFC4588 packet
2. out-of-order, too late (*), RTP media packet for non-RFC4588-enabled and NACKs-enabled RTP streams

(*) The late evaluation is done with a heuristic based on the difference between the last in-order RTP timestamp and the out-of-order packet RTP timestamp

- Jitter is now calculated only for in-order packet.

- In link quality estimation has been fixed in order to consider only the difference between `expected` and `received`

- In media link quality has been fixed in order to consider also the `retransmitted` packet.

### CODE REVIEW

**Comment by lminiero** [comment]
Pinging @pitkonst as this probably has an impact on how he originally conceived the media/link quality contribution.

**Comment by pitkonst** [comment]
Thank you for maintaining this feature. We have experienced problems with measurements in new versions of Janus. I'll check and test this changes ASAP.

**Comment by pitkonst** [comment]
I've checked inbound link quality code and evereything is fine. Tests shows that mesurments are accurate. `received` and `retransmitted` semantics changed since my initial commit, and there were problems in calculations. And this PR fixes this. @atoppi, thank you for this one.

**Comment by lminiero** [comment]
Looks good, then, merging :+1:
